pkg/cli/cmd/resource/delete/delete.go

```diff
@@ -210,13 +214,23 @@ func (r *Runner) extractEnvironmentAndApplicationIDs(ctx context.Context, client
 		}
 	}
 
+	// Detect case 4: (no environment or application)
+	if environmentID.IsEmpty() && applicationID.IsEmpty() {
+		return resources.ID{}, resources.ID{}, nil
+	}
+
 	// At this point we have the environment and application IDs **if** they were returned by
 	// the API. That covers case 1 & 2. Now we need to handle case 3, by doing an additional
 	// lookup.
 	if !environmentID.IsEmpty() {
 		return environmentID, applicationID, nil // Case 1 or Case 2
 	}
 
+	// TODO:
```

Comment:
Is this a TODO that will be handled in this PR?

Reply:
I should remove the TODO, I already fixed it.